### ISSUE TITLE
Unmatched route (Page not found) when doing from tab to another screen

### ISSUE THREAD

**Issue by Albert-Gao** [expo/router#576]
### Which package manager are you using? (Yarn is recommended)

npm

### Summary

here is the structre

---Stack
------(main-tabs)
---------tab1
------clip
---------[clipId]

You can see that (main-tabs) and clip sits at the same level

when I go to `[clipId]` from `tab1`, I got that error

### Minimal reproducible example

Here is the demo (expo-router: 1.7.1):

https://github.com/Albert-Gao/expo-router-dynamic-route

**Comment by Albert-Gao** [comment]
I kind of found the issue:

It seems the metro is not detecting any changes in the /app folder, no matter which project, I tried clear the watchman cache, pass -c to expo start, remove yarn.lock and reinstall, nothing works

metro doc says pass --reset-cache param when running, but this parameter has removed, tried updating my metro config to look like this, as the doc suggests, still not working...

```javascript
module.exports = { ...getDefaultConfig(__dirname), resetCache: true };
```

**Comment by smalik02** [comment]
Yeah - I cannot get it to work either. Have cleaned the project & rebuilt many times with no luck. Tested with adding dummy files in the root path under the app folder, but no routing luck. All giving the 404 error

**Comment by smalik02** [comment]
Nvm - this was due to not exporting function as default. Rough

**Comment by Albert-Gao** [comment]
@smalik02 thanks, mate, that's a hard one😅

Must be an error when copying and pasting!

**Comment by Albert-Gao** [comment]
Consider the human error is hard to eliminate but mitigate,
Can we have a better error message in this case, which we can read as: 

**file A in app folder has no default export, do you miss it?**

Which would make it super easy to debug.

@EvanBacon @marklawlor

**Comment by sregg** [comment]
I also lost about 1hr because of this.
It's very common to export React component using the named arrow function 
```
export const Home = () => {...}
```
instead of 
```
export default Home() {...}
```
It would be great to either warn in the logs/app or allow for named exports.
Happy to help with a PR if needed.

**Comment by ierdss** [comment]
> I also lost about 1hr because of this. It's very common to export React component using the named arrow function
> 
> ```
> export const Home = () => {...}
> ```
> 
> instead of
> 
> ```
> export default Home() {...}
> ```
> 
> I know it's written in [the docs](https://docs.expo.dev/routing/create-pages/) <img alt="image" width="510" src="https://private-user-images.githubusercontent.com/1479283/275330549-902bca54-f1db-4323-b0f9-c30e1eccd3e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc1NDgxODAsIm5iZiI6MTcwNzU0Nzg4MCwicGF0aCI6Ii8xNDc5MjgzLzI3NTMzMDU0OS05MDJiY2E1NC1mMWRiLTQzMjMtYjBmOS1jMzBlMWVjY2QzZTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTBUMDY1MTIwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTNlYWI2M2FlOTBlZmNmMDY3YThiODBiZTdkNDllZTRkNzNjYjJiZjRjMmFmZTE0MTdhMmU5ZWYxMDBiNWVkNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.NABTzGGdG-tza5eTMG0TIVJUi4VnZ7HBhTLsmMbtC8o"> but it would be great to either warn in the logs/app or allow for named exports. Happy to help with a PR if needed.

Thanks! This fixed my error. I forgot to write default for my home route.

**Comment by BalogunofAfrica** [comment]
Lost close to a day of debuging, a warning at least would help
